File: Task_1/task_1.php
<?php

function isCyclicShift($s, $t) {
    // Проверяем, равны ли длины строк
    if (strlen($s) !== strlen($t)) {
        return false;
    }

    // Создаем строку S+S
    $doubledS = $s . $s;

    // Проверяем, содержится ли T в S+S
    return strpos($doubledS, $t) !== false;
}

// Считываем строки S и T с клавиатуры
echo "Введите строку S:\n";
$s = trim(fgets(STDIN));
if ($s === false || $s === '') {
    echo "Ошибка при вводе строки S.\n";
    exit;
}

echo "Введите строку T:\n";
$t = trim(fgets(STDIN));
if ($t === false || $t === '') {
    echo "Ошибка при вводе строки T.\n";
    exit;
}

// Проверяем, является ли T циклическим сдвигом S
if (isCyclicShift($s, $t)) {
    echo "Yes\n";
} else {
    echo "No\n";
}

File: Task_1/task_1.rb
def is_cyclic_shift(s, t)
    # Проверяем, равны ли длины строк
    return false if s.length != t.length
  
    # Создаем строку S+S
    doubled_s = s + s
  
    # Проверяем, содержится ли T в S+S
    doubled_s.include?(t)
  end
  
  # Основная программа
  puts "Введите строку S:"
  s = gets.chomp
  if s.empty?
    puts "Ошибка при вводе строки S."
    exit
  end
  
  puts "Введите строку T:"
  t = gets.chomp
  if t.empty?
    puts "Ошибка при вводе строки T."
    exit
  end
  
  # Проверяем, является ли T циклическим сдвигом S
  if is_cyclic_shift(s, t)
    puts "Yes"
  else
    puts "No"
  end